### PR TITLE
upgrade purescript-react and use purescript-react-dom

### PR DESCRIPTION
**Pull request by justinwoo**
introduces package.json to install react dependencies and
make testing easier.

also shims CommonJS require() for purescript-react and react-dom.

effectively upgrades ps-thermite to react 0.14.x. please let me know if you don't want some of these changes, i can remove them as needed.

### CODE REVIEW

**Comment by pkamenarsky** [comment]
:+1: 

The current version of `thermite/react` throws warnings with newer versions of React, like:

    Warning: input is a void element tag and must not have `children` or use `props.dangerouslySetInnerHTML`. Check the render method of null.

Maybe this will fix it, if not then I'll submit a pull request for `purescript-react`.

File: html/index.html [hunk 1]
```diff
@@ -10,7 +10,17 @@
 </head>
 <body>
   <div id="container" class="container">
-    <script src="../bower_components/react/react.js"></script>
+    <script src="../node_modules/react/dist/react.js"></script>
+    <script src="../node_modules/react-dom/dist/react-dom.js"></script>
+    <script type="text/javascript">
+      function require(name) {
```
Comment (paf31) [inline html/index.html]: Does it make sense to use `browserify` or `webpack` or something here? I'm not sure if Pulp supports that for the `test` directory though.

**Comment by paf31** [comment]
Looks good! Sorry for the delay, I'll test this out later today and put out a release.

File: html/index.html (hunk 1 above)
Comment (justinwoo) [inline html/index.html]: Could be nice, but probably can't do it with vanilla pulp-test, I think (iirc it never calls browserify to do JS transforms -- I ran into an issue where I wrote my native code in ES6 and was doing babel6 transforms in browserify and test didn't pick it up)

Comment (paf31) [inline html/index.html]: Makes sense, thanks.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "purescript-thermite",
+  "version": "1.0.0",
+  "description": "[![Pursuit](http://pursuit.purescript.org/packages/purescript-thermite/badge)](http://pursuit.purescript.org/packages/purescript-thermite/)",
+  "scripts": {
+    "pretest": "pulp build",
+    "test": "pulp test -r cat > html/index.js && open html/index.html"
```
Comment (paf31) [inline package.json]: `-r` is now `--engine` as of Pulp 6.

Comment (paf31) [inline package.json]: Could you please add a line to the readme to explain that `npm run test` can now be used to run the example?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "purescript-thermite",
+  "version": "1.0.0",
+  "description": "[![Pursuit](http://pursuit.purescript.org/packages/purescript-thermite/badge)](http://pursuit.purescript.org/packages/purescript-thermite/)",
```
Comment (paf31) [inline package.json]: Is this supposed to be Markdown? Was it autogenerated?

Comment (justinwoo) [inline package.json]: Autogenerated, I can rip it out if it's ugly.

Comment (paf31) [inline package.json]: ah ok, yes please :smile:

File: package.json (hunk 2 above)
Comment (justinwoo) [inline package.json]: Updated, but didn't understand the `--engine` part, afaik I only see `-r` and `--runtime` in pulp 8.0.0

**Comment by paf31** [comment]
Thanks!

**Comment by justinwoo** [comment]
:beers:

**Pull request merged**